### PR TITLE
feat(notice): make component responsive

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #6779 

## Summary

Updates notice to be responsive per the design spec.

## Additional changes

* simplified class names (`notice` prefix is no longer necessary)
* `requestedIcon` is now computed at render time to simplify component 
* clean up test story

### CODE REVIEW

File: packages/calcite-components/src/components/notice/notice.scss [hunk 1]
```diff
@@ -68,25 +77,27 @@
 :host {
   @apply mx-auto hidden max-w-full items-center;
   inline-size: var(--calcite-notice-width);
+  container: host / inline-size;
```
Comment (driskull) [inline packages/calcite-components/src/components/notice/notice.scss:80]: I like `responsive-container / inline-size`. I think that would work well for every component.

File: packages/calcite-components/src/components/notice/resources.ts [hunk 2]
```diff
@@ -10,5 +10,8 @@ export const CSS = {
   close: "notice-close",
```
Comment (driskull) [inline packages/calcite-components/src/components/notice/resources.ts:10]: This class needs to be updated.

File: packages/calcite-components/src/components/notice/resources.ts [hunk 3]
```diff
@@ -10,5 +10,8 @@ export const CSS = {
...
   container: "container",
   content: "notice-content",
+  contentContainer: "content-container",
+  footer: "footer",
   icon: "notice-icon",
```
Comment (driskull) [inline packages/calcite-components/src/components/notice/resources.ts:15]: This class needs to be updated.

**Review by driskull — CHANGES_REQUESTED** [review]

**Review by driskull — APPROVED** [review]
Code changes look good!

**Comment by jcfranco** [comment]
@macandcheese @ashetland @SkyeSeitz Could you review the [visual changes](https://www.chromatic.com/build?appId=6266d45509d7eb004aa254fb&number=2513)? 🙇

**Comment by jcfranco** [comment]
Closing per https://github.com/Esri/calcite-design-system/issues/6779#issuecomment-1730745134.

**Pull request closed without merging**